scripts/build/build/targets.py

```diff
@@ -266,6 +266,7 @@ def BuildAndroidTarget():
         TargetPart('chip-test', app=AndroidApp.CHIP_TEST),
         TargetPart('tv-server', app=AndroidApp.TV_SERVER),
         TargetPart('tv-casting-app', app=AndroidApp.TV_CASTING_APP),
+        TargetPart('java-matter-controller', app=AndroidApp.JAVA_MATTER_CONTROLLER),
```

Comment:
Should this be for android? I would find a CLI more useful for linux to run locally. Or will we support both (like android CLI and linux CLI)?

if we compile for linux, we should probably have some separate build target/rules so we do not compile for all of arm/arm64/x86/x64

Reply:
We leverage android x86 for the first version, after https://github.com/project-chip/connectedhomeip/issues/23133, we should be able to migrate to build pure Java app with JDK

Reply:
I also wondering if we need to migrate to JDK since our goal is to validate the Matter Java Android library, no products use Matter Java library on desktop except this app.  It seems android x86 could already meet this requirement, and both java-matter-controller and Matter Android app use the same Matter lib built from same Android SDK.

Reply:
Since we can link CHIPController.jar and run the Java application built with android binary x84 directly on Linux, we plan to continue this route at this moment until we hit the problem

Reply:
I think we need to get rid of android component, it seems prebuild x86_64 android.jar can only run inside emulator, and cannot work directly in our debian machine. I think we could provide the dummy implementation for log and ble used in ChipDeviceController.Java so that we get the pure java application without android.